apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/PSSetterDefinitionOfJDBCInstrumentation.java

```diff
@@ -40,11 +40,10 @@ public PSSetterDefinitionOfJDBCInstrumentation(boolean ignorable) {
     public ElementMatcher<MethodDescription> getMethodsMatcher() {
         ElementMatcher.Junction<MethodDescription> matcher = none();
 
-        if (JDBCPluginConfig.Plugin.JDBC.TRACE_SQL_PARAMETERS) {
-            final Set<String> setters = ignorable ? PS_IGNORABLE_SETTERS : PS_SETTERS;
-            for (String setter : setters) {
-                matcher = matcher.or(named(setter));
-            }
+        // remove TRACE_SQL_PARAMETERS judgement for dynamic config
```

Comment:
Static should be kept, dynamic should only override the default value.

Reply:
There is a problem, `org.apache.skywalking.apm.plugin.jdbc.PSSetterDefinitionOfJDBCInstrumentation#getMethodsMatcher` only invoked once when instrument occur. if dynamic change later, it won't be affected, and `org.apache.skywalking.apm.plugin.jdbc.JDBCPreparedStatementSetterInterceptor#beforeMethod` will not be active even when the `TRACE_SQL_PARAMETERS` is true.

Reply:
Watcher should be initialized in the driver's interceptor, not in the instrumentation class, no matter in any case. I think the class loader issues you faced are because of this.

Reply:
![image](https://user-images.githubusercontent.com/22634485/168981639-b9904b6b-8eb4-48f3-8dd3-a53ece99b847.png)

`org.apache.skywalking.apm.plugin.jdbc.PSSetterDefinitionOfJDBCInstrumentation#getMethodsMatcher` only occur once.  Whose interceptor  `org.apache.skywalking.apm.plugin.jdbc.JDBCPreparedStatementSetterInterceptor#beforeMethod` put sql parameter into `StatementEnhanceInfos `. If do not enhance these class, even though the dynamic config changed from false to true. `org.apache.skywalking.apm.plugin.jdbc.mysql.PreparedStatementExecuteMethodsInterceptor#beforeMethod` cannot get parameter in line `final Object[] parameters = cacheObject.getParameters();`

Reply:
Don't do anything in `PSSetterDefinitionOfJDBCInstrumentation`, which is in another class loader and only executed in the booting stage.

Try to initial watcher in `JDBCPreparedStatementSetterInterceptor`

Reply:
> Don't do anything in `PSSetterDefinitionOfJDBCInstrumentation`, which is in another class loader and only executed in the booting stage.
> 
> Try to initial watcher in `JDBCPreparedStatementSetterInterceptor`

Yes, just because of `PSSetterDefinitionOfJDBCInstrumentation` only executed in the booting stage, it just executed once. But the `getMethodsMatcher` in  `PSSetterDefinitionOfJDBCInstrumentation` in the booting stage check the `JDBCPluginConfig.Plugin.JDBC.TRACE_SQL_PARAMETERS`. if it is false, `JDBCPreparedStatementSetterInterceptor` won't be active because no set method in `PreparedStatement` will be matched.

![image](https://user-images.githubusercontent.com/22634485/169222394-60f105ce-768f-4886-ae8f-ee42415f289a.png)

Reply:
You are asking a dynamic configuration, of course, you have to always instrument the target class. `PSSetterDefinitionOfJDBCInstrumentation` has to be changed. But this should not be the reason you are adding more things to the core level.
The core is unexpected to change in this case.